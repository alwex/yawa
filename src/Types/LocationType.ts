enum LocationType {
  street = 'street',
  subcity = 'subcity',
  city = 'city',
  county = 'county',
  state = 'state',
}

export default LocationType
