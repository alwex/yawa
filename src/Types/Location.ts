import LocationType from './LocationType'

interface Location {
  alternative_names: string
  boundingbox: [number, number, number, number]
  display_name: string
  lon: number
  lat: number
  name: string
  name_suffix: string
  type: LocationType
}

export default Location
