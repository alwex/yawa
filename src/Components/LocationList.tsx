import React from 'react'
import { Body, Icon, List, ListItem, Right, Text } from 'native-base'
import Location from '../Types/Location'

interface LocationListProps {
  locations: Location[]
  onLocationPress: (location: Location) => void
}

export default class LocationList extends React.Component<LocationListProps> {
  render() {
    const { locations, onLocationPress } = this.props
    return (
      <List>
        {locations.map(location => (
          <ListItem onPress={() => onLocationPress(location)} key={location.display_name}>
            <Body>
              <Text>{location.name}</Text>
              <Text note>{location.name_suffix}</Text>
            </Body>
            <Right>
              <Icon name="arrow-forward" />
            </Right>
          </ListItem>
        ))}
      </List>
    )
  }
}
