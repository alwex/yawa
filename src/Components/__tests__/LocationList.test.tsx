import React from 'react'
import renderer from 'react-test-renderer'
import LocationList from '../LocationList'
import Location from '../../Types/Location'
import LocationType from '../../Types/LocationType'

describe('LocationList', () => {
  it('render correctly', () => {
    const location1: Location = {
      alternative_names: 'Dunedin',
      boundingbox: [-77.4369278, 37.2065392, -77.4369278, 37.2065392],
      display_name: 'Dunedin, Petersburg, Virginia',
      lon: -77.4369278,
      lat: 37.2065392,
      name: 'Dunedin',
      name_suffix: 'Petersburg, Virginia',
      type: LocationType.city,
    }

    const location2: Location = {
      alternative_names: 'Dunedin',
      boundingbox: [-5.3451748, 55.5609028, -5.3451748, 55.5609028],
      display_name: 'Dunedin, North Ayrshire, Scotland',
      lon: -5.3451748,
      lat: 55.5609028,
      name: 'Dunedin',
      name_suffix: 'North Ayrshire, Scotland',
      type: LocationType.county,
    }

    const locations = [location1, location2]

    const component = renderer
      .create(<LocationList locations={locations} onLocationPress={() => {}} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
