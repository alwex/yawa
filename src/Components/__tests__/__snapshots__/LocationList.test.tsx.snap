// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationList render correctly 1`] = `
<View
  closeOnRowBeginSwipe={false}
  closeOnRowPress={true}
  closeOnScroll={true}
  directionalDistanceChangeThreshold={2}
  disableLeftSwipe={false}
  disableRightSwipe={false}
  leftOpenValue={0}
  previewFirstRow={false}
  recalculateHiddenLayout={false}
  rightOpenValue={0}
  style={Object {}}
  swipeToOpenPercent={50}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": 13,
          "paddingRight": 16,
          "paddingTop": 13,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": null,
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
          uppercase={false}
        >
          Dunedin
        </Text>
        <Text
          note={true}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
          uppercase={false}
        >
          Petersburg, Virginia
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 0.28,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": null,
            "paddingTop": null,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "alignSelf": null,
                  "color": "#c9c8cd",
                  "fontSize": 22,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": 13,
          "paddingRight": 16,
          "paddingTop": 13,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": null,
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
          uppercase={false}
        >
          Dunedin
        </Text>
        <Text
          note={true}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
          uppercase={false}
        >
          North Ayrshire, Scotland
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 0.28,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": null,
            "paddingTop": null,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "alignSelf": null,
                  "color": "#c9c8cd",
                  "fontSize": 22,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
